ticks on colorbar are not rendering properly when ordinal color scale is used

Code snippet to reproduce the issue:

```python
from bqplot import *
import numpy as np

n = 5
size_data = np.random.rand(n)
color_data = np.arange(n)

cs = OrdinalColorScale(scheme='Reds')
cax = ColorAxis(scale=cs)

pie = Pie(sizes=size_data, scales={'color': cs}, color=color_data)
Figure(marks=[pie], axes=[cax])
```
In the console I see the following error:<br>
d3.js:670 Error: <g> attribute transform: Expected number, "translate(NaN,0)"

![image](https://user-images.githubusercontent.com/5638994/50179816-03b59700-02d6-11e9-926d-137e8eaa63b9.png)

--- a/js/src/ColorAxis.js
+++ b/js/src/ColorAxis.js
@@ -332,12 +332,16 @@ var ColorBar = axis.Axis.extend({
 
     set_axisline_domain: function() {
         var domain = this.axis_scale.scale.domain();
-        var mid = this.axis_scale.model.mid;
-        if (mid === undefined || mid === null) {
-            this.axis_line_scale.domain([domain[0], domain[domain.length-1]]);
+        if (this.ordinal) {
+            this.axis_line_scale.domain(domain);
         } else {
-            this.axis_line_scale.domain([domain[0], mid, domain[domain.length-1]]);
-        }
+            var mid = this.axis_scale.model.mid;
+            if (mid === undefined || mid === null) {
+                this.axis_line_scale.domain([domain[0], domain[domain.length-1]]);
+            } else {
+                this.axis_line_scale.domain([domain[0], mid, domain[domain.length-1]]);
+            }
+        } 
     },
 
     redraw_axis: function() {

